pointer-masking-proposal.adoc

```diff
@@ -62,76 +62,87 @@ _**MPMMASK**_ layout can be found in Figure 2a, 2b, 2c and 2d for M, HS/S, VS an
 _**MPMBASE**_ layout can be found in Figure 3a, 3b, 3c and 3d for M, HS/S, VS and (V)U-mode. +
 
 Table 1 explains the meaning of _PM_ bits for RV32, RV64 and RV128. +
+Table 2 explains the meaning of _XS_ bits for RV32, RV64 and RV128. +
 
 :table-caption!:
 
 [%header, cols=5*]
-.Figure 1a: Memory Tagging Extension register (_**mmte**_) M-MODE
+.Figure 1a: Memory Tagging Extension register (_**mmte**_) M-mode
 ,===
-mmte[XLEN-1:6],mmte[5:4],mmte[3:2],mmte[1:0]
-WPRI,M-mode PM,HS/S-mode PM,(V)U-mode PM
+mmte[XLEN-1:7],mmte[6:6],mmte[5:4],mmte[3:2],mmte[1:0]
+WPRI,M-mode PM.Enabled,HS/S-mode PM,(V)U-mode PM,XS bits
 ,===
 
 [%header, cols=4*]
 .Figure 1b: Memory Tagging Extension register (_**smte**_) HS/S-mode
 ,===
-smte[XLEN-1:4],smte[3:2],smte[1:0]
-WPRI,HS/S-mode PM,U-mode PM
+smte[XLEN-1:6],smte[5:4],smte[3:2],smte[1:0]
+WPRI,HS/S-mode PM,U-mode PM,XS bits
 ,===
 
-[%header, cols=3*]
+[%header, cols=4*]
 .Figure 1c: Memory Tagging Extension register (_**vsmte**_) for VS-mode
 ,===
-vsmte[XLEN-1:4],vsmte[3:2],vsmte[1:0]
-WPRI,VS-mode PM,VU-mode PM
+vsmte[XLEN-1:6],vsmte[5:4],vsmte[3:2],vsmte[1:0]
+WPRI,VS-mode PM,VU-mode PM,XS bits
 ,===
 
-[%header, cols=2*]
+[%header, cols=3*]
 .Figure 1d: Memory Tagging Extension register (_**umte**_) for (V)U-mode
 ,===
-umte[XLEN-1:2],umte[1:0]
-WPRI,(V)U-mode PM
+umte[XLEN-1:4],umte[3:2],umte[1:0]
+WPRI,(V)U-mode PM,WPRI
 ,===
 
 [cols="15%,15%,70%", options="header"]
 .Table 1: Meaning of _PM_ bits for RV32, RV64 and RV128
 |===========================================================================================================================
 ^|*PM Bits* ^|*Name* ^|*Meaning*
-^.^| PM[0:0] ^.^| PM.Enabled |
+^.^| PM[2:0] ^.^| PM.Enabled |
 
   0 – PM is disabled (_default_) +
   1 – PM is enabled
 
-^.^| PM[1:1] ^.^| PM.Current |
+^.^| PM[3:1] ^.^| PM.Current |
 
  0 – _**xPMMASK**_ and _**xPMBASE**_ registers can only be modified by the higher privilege mode +
  1 – _**xPMMASK**_ and _**xPMBASE**_ registers can be modified by the same privilege mode
 
 |===========================================================================================================================
 
+[cols="15%,70%", options="header"]
+.Table 2: Meaning of _XS_ bits for RV32, RV64 and RV128
```

Comment:
I think we should try to keep these analogous to the definition in the privileged spec:

0 = PM extension is not used (cannot write any PM CSRs)
1 = PM CSRs are writeable but have the initial value (0)
2 = Some registers may have non-initial values, but those were saved before (clean)
3 = Some registers have unsaved values (dirty)

We could also add additional bits to contain individual dirty bits for xMTE, xPMMASK and xPMBASE.

Reply:
I've just changed XS bit definition to reflect your comments.
I did not add extra bits to track individual PM CSRs so far. For that reason I'm keeping this comment open for now.